packages/box/src/Box.tsx

```diff
@@ -1,33 +1,59 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-type BoxDivDefault = {
```

Comment:
Mind summarizing the changes in this file? It seems like a lot.

Reply:
The main change was to make `BoxProps` be generic so clients could pass in their own overrides. Most of the rest of the changes are drilling the overrides down into each overload variant. I also added the `Override` helper to prevent issues where clients might be using `as` or `href` for their own purposes.